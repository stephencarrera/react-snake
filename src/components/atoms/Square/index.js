import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Square = ({ backgroundColor }) => (
  <StyledSquare backgroundColor={backgroundColor} />
);

export const StyledSquare = styled.div`
  width: '20px';
  height: '20px';
  margin: '0px';
  display: 'inline-block';
  background: ${props => props.backgroundColor};
`;

Square.propTypes = {
  backgroundColor: PropTypes.string
};

export default Square;
