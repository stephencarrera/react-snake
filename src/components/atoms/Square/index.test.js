import React from 'react';
import Square, { StyledSquare } from './';
import { shallow } from 'enzyme';

it('Square renders without props', () => {
  const wrapper = shallow(<Square />);
  expect(wrapper).toMatchSnapshot();
});

it('StyledSquare matches its snapshot with color', () => {
  const wrapper = shallow(<StyledSquare backgroundColor="black" />);
  expect(wrapper).toMatchSnapshot();
});
