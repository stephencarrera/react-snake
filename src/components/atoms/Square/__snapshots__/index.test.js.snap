// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square renders without props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Square />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "backgroundColor": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "backgroundColor": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`StyledSquare matches its snapshot with color 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div
    backgroundColor="black"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-bdVaJa gfgvM",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-bdVaJa gfgvM",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
